### DISCUSSION TITLE
Welcome to RfemPythonWsClient Discussions!

### DISCUSSION THREAD

**Discussion by jarabroz** [Dlubal-Software/RFEM_Python_Client#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by MGeesink** [comment]
Hello,

At the "NodalSupportType" you can choose from a number of options (Hinged, Fixed, Roller etc) which determine the degrees of freedom of the support. Is it possible to program the support where you can define the degrees of freedom yourself.
Example: ux = free, uy = fixed, uz = fixed and φx = fixed, φy = free, φz = fixed.

**Reply by AshkanSh1990** [reply]
Hi @MGeesink,

I believe there's an alternative solution available as well, which involves creating a list instead of utilizing an enum for predefined boundary conditions. To implement this approach, you should import the following class and structure the nodal support as follows:

from RFEM.dataTypes import inf

NodalSupport(1,"5",[0,inf,inf,inf,0,inf])

**Comment by OndraMichal** [comment]
Hi @MGeesink,
generaly you can set any parameter by utilizing dictionary `params` which is optional parameter in almost every method. If you you are interested in only NodalSupport, this object/class has dedicated setNodalSupportConditions class to specify any configuration of nodal support using C_u_X-Z and C_phi_X-Z. See nodalSupporty.py. 
Did I answered your question?

**Comment by bogdanvadan** [comment]
Hello,

I just started to use the RFEM python client and I really like it, but I got to a point where I can`t set to the surfaces the input axes. I found in the javascript documentation that this could be done, but in the python I didn`t found this option

Thank you!

**Reply by bogdanvadan** [reply]
Thank you @OndraMichal I got it working.

Further with my script I got to some other small 

1.  Assign member support
    MemberSupport(int(k/2+1), f"{int(k/2+1)}",  0,   10000, [10000, MemberSupportNonlinearity.NONLINEARITY_NONE], 0, 0 , 0 )
    I get this error:  "LocalCache\local-packages\Python311\site-packages\RFEM\TypesForMembers\memberSupport.py", line 52, in __init__
    clientObject.nonlinearity = spring_translation_z[1].name
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'name' "

2. For nodal support, if I get this from the get_nodal_support:
   "no = 1
   user_defined_name_enabled = False
   name = "EEN --- | C_u,x : 10.00 kN/m | C_u,y : 10.00 kN/m | C_u,z : 1000.00 kN/m | u_z : Diagram (Nodes : 2641)"
   nodes = "2641"
   (...)
   spring_z_nonlinearity = "NONLINEARITY_TYPE_DIAGRAM"
   diagram_along_z_end = "DIAGRAM_ENDING_TYPE_YIELDING"
   diagram_along_z_is_sorted = True
   diagram_along_z_start = "DIAGRAM_ENDING_TYPE_FAILURE"
   diagram_along_z_symmetric = False
   diagram_along_z_table =
      (array_of_nodal_support_diagram_along_z_table){
         nodal_support_diagram_along_z_table[] =
            (nodal_support_diagram_along_z_table_row){
               no = 1
               description = None
               row =
                  (nodal_support_diagram_along_z_table){
                     displacement = -0.008
                     force = -30000000.0
                     spring = 5000000000.0
                     note = None
                  }
            },
            (nodal_support_diagram_along_z_table_row){
               no = 2
               description = None
               row =
                  (nodal_support_diagram_along_z_table){
                     displacement = -0.006
                     force = -20000000.0
                     spring = 5000000000.0
                     note = None
                  }
            },
            (nodal_support_diagram_along_z_table_row){
               no = 3
               description = None
               row =
                  (nodal_support_diagram_along_z_table){
                     displacement = -0.005
                     force = -15000000.0
                     spring = 1666666666.6666667
                     note = None
                  }
            }, ..... "

How would I define the parameters dictionary/list for the diagram along z table?

3. Is there a way to extract the results from the ResultSections?

**Comment by OndraMichal** [comment]
Hi @bogdanvadan,
thank you for the interest. Take a look at [this](https://github.com/Dlubal-Software/RFEM_Python_Client/wiki/How-to-set-'params').
